### PR TITLE
build(docker): Add docker setup for development

### PR DESCRIPTION
**Pull request by tcrosado**
Now the development setup can be done using a docker container (based on [lead4good/truffle](https://hub.docker.com/r/lead4good/truffle/) image).
To build the image you should execute:
`docker-compose build` 
In order to start the container, execute:
`docker-compose up`
Wait a bit for ganache to display develpment addresses and later execute on a separate terminal:
`docker exec -it AppCoinsDev  bash` 

This will open a bash shell on the docker container where you can execute the usual development commands such as `truffle compile`.

Related to this pull request I made some decisions that should be reviewed:
 - ganache-cli is running on debug mode (is it necessary?)
 - docker exec commands are running as root on the container, should it be changed to an unprivileged user?
- should documentation about the docker setup be added to this PR or add it to a new PR later?

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
+FROM node:10.5.0-alpine
```
Comment (DioPires) [inline Dockerfile:1]: #### Tests:
```diff 
+ 30 passing
```
#### Test coverage: 
81.94 %

Comment (DioPires) [inline Dockerfile:1]: #### Tests:
```diff 
+ 30 passing
```
#### Test coverage: 
81.94 %

**Review by GustavoRSSilva — APPROVED** [review]

**Comment by tcrosado** [comment]
ganache-cli is now running in normal mode and instructions for setup were added

File: Dockerfile (hunk 1 above)
Comment (DioPires) [inline Dockerfile:1]: #### Tests:
```diff 
+ 30 passing
```
#### Test coverage: 
81.94 %

Comment (DioPires) [inline Dockerfile:1]: #### Tests:
```diff 
+ 30 passing
```
#### Test coverage: 
81.94 %

**Pull request merged**